### ISSUE TITLE
Is the trunk pretrained as a decoder-only model, like GPT with autoregressive generation and causal mask?

### ISSUE THREAD

**Issue by UniGen-X** [liruiw/HPT#34]
Hi,

We would like to double-check if the trunk in the released pre-trained checkpoints was trained via a decoder-only way, with causal attention mask and autoregressive generation? 

The appendix mentioned that "**The trunk is parametrized by a decoder-only transformer architecture**", however, when running the code, we noticed that the attn_mask in the forward function of [SimpleTransformer](https://github.com/liruiw/HPT/blob/549433f70569917eab84dfe0eae9d84a7c4ad9a3/hpt/models/transformer.py#L232) was always```NONE```.

Thanks!

**Comment by liruiw** [comment]
The trunk is not trained with causal attention mask and autoregresive generation.
